web-common/src/layout/overlay-store.ts

```diff
@@ -11,4 +9,32 @@ interface Overlay {
   };
 }
 
-export const overlay = writable<Overlay | null>(null);
+const { subscribe, set } = writable<Overlay | null>(null);
+let timeout: NodeJS.Timeout;
+let isCleared: boolean = false;
```

Comment:
A class works best here since we have variables local to this. Perhaps good to make it a util so that it could be used elsewhere as well?